pennylane_lightning_gpu/src/simulator/StateVectorCudaMPI.hpp

```diff
@@ -64,6 +64,25 @@ extern void setBasisState_CUDA(cuDoubleComplex *sv, cuDoubleComplex &value,
                                const size_t index, bool async,
                                cudaStream_t stream_id);
 
+template <class Precision, class index_type> class CSRMatrix {
+  private:
+    std::vector<index_type> columns;
+    std::vector<index_type> csrOffsets;
+    std::vector<std::complex<Precision>> values;
+
+  public:
+    CSRMatrix(size_t num_rows, size_t nnz)
+        : columns(nnz, 0), csrOffsets(num_rows + 1, 0), values(nnz){};
+
+    CSRMatrix() = default;
+
+    auto getColumns() -> std::vector<index_type> & { return columns; }
+    auto getValues() -> std::vector<std::complex<Precision>> & {
+        return values;
+    }
+    auto getCsrOffsets() -> std::vector<index_type> & { return csrOffsets; }
```

Comment:
And docstrings to explain what these return --- mainly so anybody unfamiliar with CSR can identify how to build a matrix using these.

Reply:
Done!